src/symbols_linux.cpp

```diff
@@ -739,35 +741,20 @@ static int parseLibrariesCallback(struct dl_phdr_info* info, size_t size, void*
             break;
         }
 
-        cc = new CodeCache(map.file(), count, false, map_start, map_end);
-        cc_inode = inode;
+        // If last_inode is set, image_base is known to be valid and readable
+        ImageBaseStatus imageBaseStatus = inode != 0 && inode == last_inode
+            ? IMAGE_BASE_VALID : (unsigned long)map_start > map_offs
+            ? IMAGE_BASE_NOT_FOUND : IMAGE_BASE_STATUS_UNKNOWN;
 
-        if (strchr(map.file(), ':') != NULL) {
-            // Do not try to parse pseudofiles like anon_inode:name, /memfd:name
-        } else if (inode != 0) {
-            if (inode == last_inode) {
-                // If last_inode is set, image_base is known to be valid and readable
-                ElfParser::parseFile(cc, image_base, map.file(), true);
-                // Parse program headers after the file to ensure debug symbols are parsed first
-                ElfParser::parseProgramHeaders(cc, image_base, map_end, OS::isMusl());
-            } else if ((unsigned long)map_start > map_offs) {
-                // Unlikely case when image_base has not been found.
-                // Be careful: executable file is not always ELF, e.g. classes.jsa
-                ElfParser::parseFile(cc, map_start - map_offs, map.file(), true);
-            }
-        } else if (strcmp(map.file(), "[vdso]") == 0) {
-            ElfParser::parseProgramHeaders(cc, map_start, map_end, true);
-        }
+        // Create CodeCache, do not parse the file yet.
+        cc = new CodeCache(map.file(), count, false, map_start, map_end, map_offs, imageBaseStatus);
```

Comment:
Adding unparsed CodeCache is dangerous: other parts of the code (e.g. `patchLibraries`) assume that if a library presents in CodeCacheArray it is already parsed.